src/dotnet-core-uninstall/Windows/UninstallCommandExec.cs

```diff
@@ -31,8 +32,15 @@ public static void ExecuteUninstallCommand(IEnumerable<Bundle> bundles)
             var verbosityLevel = CommandLineConfigs.CommandLineParseResult.RootCommandResult.GetVerbosityLevel();
             var verbosityLogger = new VerbosityLogger(verbosityLevel);
 
+            var bundleMutex = new Mutex();
```

Comment:
The purpose of this mutex is to prevent the loop from continuing to the next uninstall when the cancelation happens.

Reply:
There's only one thread of execution that ever attempts to acquire the mutex, so I don't see how it would prevent that.

Reply:
Because I use the `Exited` event, I thought the loop might proceed while the event handler gets executed. But if we do that after `WaitForExit`, then I don't think we need this mutex any more.

Reply:
`Mutex` in .NET can have repeated calls on the same thread to acquire it without blocking; there must be the same number of `ReleaseMutex` calls to finally release the mutex, though.

As `Mutex` has thread-affinity, blocking on repeated acquisitions on the same thread would cause deadlock since another thread cannot release the mutex acquired by a different thread.

As at-most one thread is acquiring and releasing the mutex, this is effectively unnecessary.